test/integration/evictions/evictions_test.go

```diff
@@ -502,6 +529,113 @@ func TestEvictionWithUnhealthyPodEvictionPolicy(t *testing.T) {
 	}
 }
 
+// TestEvictionWithPrecondition tests eviction with delete preconditions
+func TestEvictionWithPrecondition(t *testing.T) {
+	cases := map[string]struct {
+		enforceResourceVersion     bool
+		injectWrongResourceVersion bool
+		enforceUID                 bool
+		injectWrongUID             bool
+		shouldErr                  bool
+	}{
+		"eviction enforcing resource version": {
+			enforceResourceVersion: true,
+		},
+		"eviction enforcing UID": {
+			enforceUID: true,
+		},
+		"eviction enforcing resource version and UID": {
+			enforceUID:             true,
+			enforceResourceVersion: true,
+		},
+		"eviction enforcing wrong resource version should fail": {
+			enforceResourceVersion:     true,
+			injectWrongResourceVersion: true,
+			shouldErr:                  true,
+		},
+		"eviction enforcing wrong UID should fail": {
+			enforceUID:     true,
+			injectWrongUID: true,
+			shouldErr:      true,
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			closeFn, rm, informers, _, clientSet := rmSetup(t)
+			defer closeFn()
+
+			ns := framework.CreateNamespaceOrDie(clientSet, "eviction-with-preconditions", t)
+			defer framework.DeleteNamespaceOrDie(clientSet, ns, t)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			informers.Start(ctx.Done())
+			go rm.Run(ctx)
+
+			pod := newPod("pod")
+			pod, err := clientSet.CoreV1().Pods(ns.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
+			if err != nil {
+				t.Errorf("Failed to create pod: %q", err)
+			}
+
+			pod.Status.Phase = v1.PodRunning
+			addPodConditionReady(pod)
+
+			// generate a new resource version
+			updatedPod, err := clientSet.CoreV1().Pods(ns.Name).UpdateStatus(ctx, pod, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			waitToObservePods(t, informers.Core().V1().Pods().Informer(), 1, v1.PodRunning)
+
+			deleteOption := metav1.DeleteOptions{}
+
+			if tc.enforceResourceVersion || tc.enforceUID {
+				deleteOption.Preconditions = &metav1.Preconditions{}
+			}
+
+			if tc.enforceResourceVersion {
+				if tc.injectWrongResourceVersion {
+					deleteOption.Preconditions.ResourceVersion = &pod.ResourceVersion
+				} else {
+					deleteOption.Preconditions.ResourceVersion = &updatedPod.ResourceVersion
+				}
+
+			}
+			if tc.enforceUID {
+				if tc.injectWrongUID {
+					newUID := uuid.NewUUID()
+					deleteOption.Preconditions.UID = &newUID
+				} else {
+					deleteOption.Preconditions.UID = &updatedPod.UID
+				}
+			}
+
+			eviction := newV1Eviction(ns.Name, updatedPod.Name, deleteOption)
+			err = clientSet.CoreV1().RESTClient().Post().
```

Comment:
can use noRetriesRESTClient here as well and call the normal clientset Evict method